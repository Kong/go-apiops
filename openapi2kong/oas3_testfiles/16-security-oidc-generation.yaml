# security directives can generate oidc plugin configs

openapi: '3.0.0'
info:
  title: OIDC tests
  version: v2
servers:
  - url: /

# defining a toplevel security spec
security:
  - doclevel-oidc: ["top-scope"]

# overriding the top level with per operation ones below
paths:
  /path1:
    get:
      operationId: OpsId1
      # this should get the plugin per doclevel-oidc specs, since we do not override
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
  /path2:
    get:
      operationId: OpsId2
      # this should get the plugin per "oidc2" specs, NO additional scopes
      summary: List API versions
      security:
        - oidc2: [] # no additional scopes
      responses:
        '200':
          description: |-
            200 response
  /path3:
    get:
      operationId: OpsId3
      # this should get the plugin per "oidc3" specs, with additional scopes
      summary: List API versions
      security:
        - oidc3: [ "extra-scope" ] # with additional scopes
      responses:
        '200':
          description: |-
            200 response
  /path4:
    get:
      operationId: OpsId4
      # this should get the plugin per "oidc4" specs, without references
      summary: List API versions
      security:
        - oidc4: [ "oidc4-scope" ]
      responses:
        '200':
          description: |-
            200 response
  /path5:
    get:
      operationId: OpsId5
      # same as no 4, but a different scope
      summary: List API versions
      security:
        - oidc4: [ "oidc5-scope" ] # different scope; 5
      responses:
        '200':
          description: |-
            200 response


components:
  x-kong:
    plugins:
      oidc2:
        config:
          run_on_preflight: true
          scopes_required: ["oidc2-plugin-ref"]
    security:
      oidc3:
        config:
          run_on_preflight: true
          scopes_required: ["oidc3-security-ref"]


  securitySchemes:
    doclevel-oidc:
      type: openIdConnect
      openIdConnectUrl: https://konghq.com/top-level
      x-kong-security-openid-connect:
        config:
          run_on_preflight: false
          scopes_required: ["scope2", "scope1"]  # reversed order to test sorting!

    oidc2:
      type: openIdConnect
      openIdConnectUrl: https://konghq.com/oidc2
      # using a reference to a security scheme, in the "plugins" section
      x-kong-security-openid-connect:
        "$ref": "#/components/x-kong/plugins/oidc2"

    oidc3:
      type: openIdConnect
      openIdConnectUrl: https://konghq.com/oidc3
      # using a reference to a security scheme, in the "security" section
      x-kong-security-openid-connect:
        "$ref": "#/components/x-kong/security/oidc3"

    oidc4:
      type: openIdConnect
      openIdConnectUrl: https://konghq.com/oidc4
      x-kong-security-openid-connect:
        config:
          run_on_preflight: false
          scopes_required: []
          issuer: "https://konghq.com/oidc4/overridden"
