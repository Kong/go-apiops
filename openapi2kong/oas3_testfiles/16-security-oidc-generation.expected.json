{
  "_format_version": "3.0",
  "services": [
    {
      "host": "localhost",
      "id": "bcda787b-18bc-5f83-85f2-1cf83b5f8eb3",
      "name": "oidc-tests",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "558eb5de-46b9-5500-ae2a-9a8e321e9d65",
          "methods": [
            "GET"
          ],
          "name": "oidc-tests_opsid1",
          "paths": [
            "~/path1$"
          ],
          "plugins": [
            {
              "config": {
                "issuer": "https://konghq.com/top-level",
                "run_on_preflight": false,
                "scopes_required": [
                  "scope1",
                  "scope2",
                  "top-scope"
                ]
              },
              "name": "openid-connect"
            }
          ],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_16-security-oidc-generation.yaml"
          ]
        },
        {
          "id": "dac8f025-ad7b-52f3-8755-8b15ab97b57e",
          "methods": [
            "GET"
          ],
          "name": "oidc-tests_opsid2",
          "paths": [
            "~/path2$"
          ],
          "plugins": [
            {
              "config": {
                "issuer": "https://konghq.com/oidc2",
                "run_on_preflight": true,
                "scopes_required": [
                  "oidc2-plugin-ref"
                ]
              },
              "name": "openid-connect"
            }
          ],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_16-security-oidc-generation.yaml"
          ]
        },
        {
          "id": "5d712419-79fa-5ed0-90fb-acd005490af5",
          "methods": [
            "GET"
          ],
          "name": "oidc-tests_opsid3",
          "paths": [
            "~/path3$"
          ],
          "plugins": [
            {
              "config": {
                "issuer": "https://konghq.com/oidc3",
                "run_on_preflight": true,
                "scopes_required": [
                  "extra-scope",
                  "oidc3-security-ref"
                ]
              },
              "name": "openid-connect"
            }
          ],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_16-security-oidc-generation.yaml"
          ]
        },
        {
          "id": "47dad727-b8ab-5bbc-93db-40c50c9e107d",
          "methods": [
            "GET"
          ],
          "name": "oidc-tests_opsid4",
          "paths": [
            "~/path4$"
          ],
          "plugins": [
            {
              "config": {
                "issuer": "https://konghq.com/oidc4",
                "run_on_preflight": false,
                "scopes_required": [
                  "oidc4-scope"
                ]
              },
              "name": "openid-connect"
            }
          ],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_16-security-oidc-generation.yaml"
          ]
        },
        {
          "id": "16cf13af-2905-5493-93fd-bbe399e2e494",
          "methods": [
            "GET"
          ],
          "name": "oidc-tests_opsid5",
          "paths": [
            "~/path5$"
          ],
          "plugins": [
            {
              "config": {
                "issuer": "https://konghq.com/oidc4",
                "run_on_preflight": false,
                "scopes_required": [
                  "oidc5-scope"
                ]
              },
              "name": "openid-connect"
            }
          ],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_16-security-oidc-generation.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_16-security-oidc-generation.yaml"
      ]
    }
  ],
  "upstreams": []
}